src/backend/dx12/src/command.rs

```diff
@@ -1213,17 +1244,128 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
+            image::Filter::Nearest => d3d12::D3D12_FILTER_MIN_MAG_MIP_POINT,
+            image::Filter::Linear => d3d12::D3D12_FILTER_MIN_MAG_LINEAR_MIP_POINT,
+        };
+
+        for region in regions {
+            let r = region.borrow();
+
+            let first_layer = r.dst_subresource.layers.start;
+            let num_layers = r.dst_subresource.layers.end - first_layer;
+
+            let blit = match r.dst_subresource.aspects {
+                format::Aspects::COLOR => {
+                    // Create RTV of the dst image for the miplevel of the current region
+                    let mut desc = d3d12::D3D12_RENDER_TARGET_VIEW_DESC {
+                        Format: dst.dxgi_format,
+                        ViewDimension: d3d12::D3D12_RTV_DIMENSION_TEXTURE2DARRAY,
+                        u: unsafe { mem::zeroed() },
+                    };
+
+                    *unsafe{ desc.u.Texture2DArray_mut() } = d3d12::D3D12_TEX2D_ARRAY_RTV {
+                        MipSlice: r.dst_subresource.level as _,
+                        FirstArraySlice: first_layer as _,
+                        ArraySize: num_layers as _,
+                        PlaneSlice: 0, //TODO
+                    };
+
+                    unsafe {
+                        (*self.shared.device).CreateRenderTargetView(dst.resource, &desc, rtv);
+                        self.raw.OMSetRenderTargets(
+                            1,
+                            &rtv,
+                            FALSE,
+                            ptr::null(),
+                        );
+                    }
+
+                    self.shared.service_pipes.get_blit_2d_color(dst.dxgi_format, filter)
+                },
+                _ => unimplemented!(),
+            };
+
+            self.set_internal_graphics_pipeline();
+
+            // TODO: support flipping?
+            let viewport = d3d12::D3D12_VIEWPORT {
+                TopLeftX: r.dst_bounds.start.x as _,
+                TopLeftY: r.dst_bounds.start.y as _,
+                Width: (r.dst_bounds.end.x - r.dst_bounds.start.x) as _,
+                Height: (r.dst_bounds.end.y - r.dst_bounds.start.y) as _,
+                MinDepth: 0.0,
+                MaxDepth: 1.0,
+            };
+            let scissor = d3d12::D3D12_RECT {
+                left: r.dst_bounds.start.x,
+                top: r.dst_bounds.start.y,
+                right: r.dst_bounds.end.x,
+                bottom: r.dst_bounds.end.y,
+            };
+
+            // TODO: update push constants
+
+            // Instanced full screen triangle
+            //
+            // Each instance corresponds to one blitting operation on one layer
+            unsafe {
+                self.raw.SetPipelineState(blit.pipeline.as_raw());
+                self.raw.SetGraphicsRootSignature(blit.signature.as_raw());
+                self.raw.RSSetViewports(1, &viewport);
+                self.raw.RSSetScissorRects(1, &scissor);
+                self.raw.DrawInstanced(
```

Comment:
where do you select that each instance goes to a particular target layer?

Reply:
Fixed/reworked